cmd/controller/ctrl_test.go

```diff
@@ -46,3 +50,15 @@ func TestDevList(t *testing.T) {
 		}
 	}
 }
+
+func TestGetnerateUid(t *testing.T) {
+	disksDetails := [2][4]string{
+		{"data-1", "data-2", "data-3", "data-4"},
+		{"testdata-1", "testdata-2", "testdata-3", "testdata-4"},
+	}
+	for _, tc := range disksDetails {
+		result := controller.GetnerateUid(tc[0], tc[1], tc[2], tc[3])
+		expected := controller.NDMPrefix + util.Hash(tc[0]+tc[1]+tc[2]+tc[3])
+		assert.Equal(t, result, expected)
```

Comment:
The expected and result are using the same method/logic. This can't test the validity of the function. For example, the intent here is to test that:
- passing the same set of values will generate the same UUID, so that a disk object can be uniquely identified across nodes
- what is the validity of this function if some parameters are blank different.

Reply:
`GetUuid()` is mainly responsible for generate uuid of a disk `GenerateUuid()` generates md5hash of 4 string. I have changed the logic now using fork exec of md5hash to get the hash.